pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+        <Identity<T>>::remove_account_key_ref_count(&user_key);
+
+        // Remove paying key for user key.
+        <Subsidies<T>>::remove(&user_key);
+
+        Self::deposit_event(RawEvent::RemovedPayingKey(
+            sender_did.for_event(),
+            user_key,
+            paying_key,
+        ));
+        Ok(())
+    }
+
+    fn base_update_polyx_limit(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: paying_key,
+            primary_did: paying_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // Check if the current paying key matches.
+        Self::ensure_is_paying_key(&user_key, &paying_key)?;
+
+        // Update polyx limit
+        <Subsidies<T>>::mutate(&user_key, |subsidy| {
+            if let Some(subsidy) = subsidy {
+                subsidy.remaining = polyx_limit;
+            }
+        });
+
+        Self::deposit_event(RawEvent::UpdatedPolyxLimit(
+            paying_did.for_event(),
+            user_key,
+            paying_key,
+            polyx_limit.into(),
+        ));
+        Ok(())
+    }
+
+    /// Adds an authorization to add a `paying_key` to the `user_key`
+    pub fn unsafe_add_auth_for_paying_key(
+        from: IdentityId,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+        polyx_limit: T::Balance,
+    ) -> u64 {
+        let auth_id = <Identity<T>>::add_auth(
+            from,
+            Signatory::Account(user_key.clone()),
+            AuthorizationData::AddRelayerPayingKey(
+                user_key.clone(),
+                paying_key.clone(),
+                polyx_limit.into(),
+            ),
+            None,
+        );
+        Self::deposit_event(RawEvent::AuthorizedPayingKey(
+            from.for_event(),
+            user_key,
+            paying_key,
+            polyx_limit.into(),
+            auth_id,
+        ));
+        auth_id
+    }
+
+    /// Check if the `key` has a valid CDD.
+    fn key_has_valid_cdd(key: &T::AccountId) -> bool {
+        if let Some(did) = <Identity<T>>::get_identity(key) {
+            <Identity<T>>::has_valid_cdd(did)
+        } else {
+            false
+        }
+    }
+
+    /// Ensure that `paying_key` is the paying key for `user_key`.
+    fn ensure_is_paying_key(user_key: &T::AccountId, paying_key: &T::AccountId) -> DispatchResult {
+        // Check if the current paying key matches
+        match <Subsidies<T>>::get(user_key) {
+            // There was no subsidy.
+            None => fail!(Error::<T>::NoPayingKey),
+            // paying key doesn't match
+            Some(s) if s.paying_key != *paying_key => fail!(Error::<T>::NotPayingKey),
+            Some(_) => Ok(()),
+        }
+    }
+
+    /// Validate and accept a `paying_key` for the `user_key`.
+    fn auth_accept_paying_key(
+        signer: Signatory<T::AccountId>,
```

Comment:
```suggestion
        user_did: IdentityId,
```
And then you can do `<Identity<T>>::has_valid_cdd(user_did)`